vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -254,21 +249,15 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 			allErrs = append(allErrs, err)
 		}
 
-		// the outermost object will be converted to the output-version, but inner
-		// objects can use their mappings
-		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
-		if err != nil {
-			return err
-		}
-
-		obj, err := resource.AsVersionedObject(infos, !singular, version, f.JSONEncoder())
-		if err != nil {
-			return err
+		for ix := range infos {
+			filter := kubectl.NewResourceFilter(filterOptionsForCommand(cmd, allNamespaces))
```

Comment:
> You have to track whether the filter removed anything to trigger "was filtered out" message.

Ah, did this upstream although since I am creating a filter per every info I had to approach it differently to how they had it. Would a better implementation be to determine if a `runtime.Object` is a list of resources, then either return a list with unfiltered pods / remove filtered pods from that list?